GetDKAN/dkan-tools
# Change behavior of restoreDb command and move namespaces; support D8/DKAN2

Fixes #66

This moves dkan:restore into its own class: RestoreCommands and consolidates some logic:

- There is no longer a -b switch in the install command.
- The dkan:restore command is now simply restore.
- A new command, restore:db, can take a URL argument, a filename inside _/backups, or will offer a selection of files inside /backups if no argument given.
- The dktl make command now makes DKAN and Drupal in one step. All make-related code is now in its own class.
- dktl dkan:deploy is now simply dktl deploy.

**QA Steps**

- Try to clone an existing site and adapt the dktl.yml file to remove the dkan: namespace from the restore command. Try dktl restore; it should work as expected.
- Create a local db dump and place it in /backups. Typing dktl restore:db should restore it.
- Create a second db dump in /backups. Typing dktl restore:db should provide a selection.
- Try creating a site from scratch. Add some overrides to drupal.make (as suggested in docs). Run dktl make and check behavior for any issues.

### Also

This PR contains some code that allow us to segregate commands. In this case we are using it to have Drupal version specific commands: D8 vs D7.

For D8, some new commands were added to get a basic DKAN2 in D8 site up and running with DKTL.

Instructions on how to stand up a DKAN2 site are available in the better-default branch in the dkan2 repo.
